src/openapi/streaming/subscription.ts

```diff
@@ -240,6 +247,33 @@ class Subscription {
         this.setState(this.STATE_UNSUBSCRIBED);
     }
 
+    /**
+     * If we get 3 resets within 1 minute then we wait for 1 minute
+     * since it may indicate some problem with publishers
+     */
+    private checkIfPublisherDown() {
+        this.resetTimeStamps.push(Date.now());
+
+        if (this.resetTimeStamps.length >= 3) {
+            this.resetTimeStamps = this.resetTimeStamps.slice(-3);
+            if (
+                !this.waitForPublisherToRespondTimer &&
+                this.resetTimeStamps[2] - this.resetTimeStamps[0] <
+                    MIN_WAIT_FOR_PUBLISHER_TO_RESPOND_MS
+            ) {
+                // 3 reset within 1 minute so wait for 1 minute for pubslisher to respond
+                log.warn(LOG_AREA, 'Received 3 resets within 1 minute');
+                this.setState(this.STATE_PUBLISHER_DOWN);
+
+                this.waitForPublisherToRespondTimer = window.setTimeout(() => {

```

Comment:
Shouldn't we empty the resetTimestamps array here too? Seems like once we have 3 timestamps we will perform this check with every reset since the array will always have 4 items? Is that expected? Am I missing sth?

Reply:
- so this function will only be called when reset happens
- we will always have latest 3 reset timestamps in this array
```
if (this.resetTimeStamps.length >= 3) {
            this.resetTimeStamps = this.resetTimeStamps.slice(-3);
```

-  we do wait for 1minute when we get 3resets within a minute and while waiting we don't start another wait 
```
!this.waitForPublisherToRespondTimer &&
                this.resetTimeStamps[2] - this.resetTimeStamps[0] <
                    MIN_WAIT_FOR_PUBLISHER_TO_RESPOND_MS
```
- after that 3 will need 3 more news resets that should happen within one minute to trigger the wait again